src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesSettings/ExternalDataSourcesSettings.js

```diff
@@ -0,0 +1,92 @@
+import { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import arrayMutators from 'final-form-arrays';
+
+import ExternalDataSourcesView from '../ExternalDataSourcesView/ExternalDataSourcesView';
+import ExternalDataSourcesLookup from '../ExternalDataSourcesLookup/ExternalDataSourcesLookup';
+import ExternalDataSourcesFormModal from '../ExternaldataSourcesFormModal/ExternalDataSourcesFormModal';
+import ExternalDataSourceForm from '../ExternalDataSourceForm/ExternalDataSourceForm';
+
+const EDITING = 'edit';
+const CREATING = 'create';
+const VIEWING = 'view';
+
+const ExternalDataSourcesSettings = ({
+  onDelete,
+  onCancel,
+  onSave,
+  externalKbs,
+  onSubmit,
+  initialValues
+}) => {
+  const [mode, setMode] = useState(VIEWING);
+  const [externalDataSource, setExternalDataSource] = useState();
+
+  return (
+    <>
+      <ExternalDataSourcesLookup
+        externalKbs={externalKbs}
+        onClick={() => setMode(CREATING)}
+        onSelectedExternalDataSource={(_e, eds) => {
+          setExternalDataSource(eds);
+        }}
+      />
+      {
+        externalDataSource &&
+          <ExternalDataSourcesView
+            externalDataSourceId={externalDataSource.id}
+            externalKbs={externalKbs}
+            initialValues={initialValues}
+            onCancel={onCancel}
+            onClick={() => setMode(EDITING)}
```

Comment:
onClick and onCancel aren't really super explanatory names here, or above.

If we end up keeping them I'd rather we passed those down as `onEdit` and `onEditCancel` or something like that.

In general I'd rather a component's props make sense in isolation, so not only in relation to the component rendering it :)